spec/lib/object_serializer_relationship_serializer_proc_spec.rb

```diff
@@ -104,4 +121,50 @@ class CarSerializer
       expect(car_included[:attributes][:model]).to eq car.model
     end
   end
+
+  context 'when serializing a polymorphic relationship with a serializer block' do
+    it 'should output the correct JSON based on the proper serializer' do
+      person = Person.new
+      person.id = 1
+      person.name = 'Bob'
+      person.polymorphic_assets = [big_house, car]
+      person_hash = PersonSerializer.new(person).to_hash
+
+      relationships = person_hash[:data][:relationships]
+      big_house_relationship = relationships[:'polymorphic-assets'][:data][0]
+      expect(big_house_relationship[:type].to_s).to eq 'house'
```

Comment:
Is this even correct? Or should the type be `big_house`, because it is `polymorphic: true`?

Reply:
@doits the `type` reflects the value of the serializer `set_type` call, so `house` seems about right:
https://github.com/fast-jsonapi/fast_jsonapi/pull/68/files#diff-20df50225c2e9086e4fd007958df5f79L34

Reply:
:+1: thanks, I wasn't sure because the relationship is specified as `polymorphic: true` so I thought it might change the `type`, but I guess I confused `record_type` (from [options in README](https://github.com/fast-jsonapi/fast_jsonapi#customizable-options))with the `type` property

Reply:
Btw, this was done by @christophersansone in #32

Reply:
I see, maybe simply moving the [check for polymorphic](https://github.com/fast-jsonapi/fast_jsonapi/pull/32/files#diff-a413c6e16d70896fa04f5fa41931cadaR70-R73) one `elsif` down after the check of `serializer` proc might fix it, if this makes sense at all

Reply:
... indeed, it makes the spec pass. But maybe it was like this on purpose, or is it better now?

Reply:
@doits please take a look at the notes in the #32 

The plan is to move away from `type`, `record_type` and `serializer` in favor of just the last one. This way we have one way to configure a relationship, just generally.

Reply:
OK, I wasn't sure. So the idea is to only have the `serializer` option in the end.

[This fix](https://github.com/fast-jsonapi/fast_jsonapi/pull/68/commits/e7d866253e78a3266f043d0fe90e9acf8ad3600d) might be correct, because it gives the `serializer` option a higher priority than the soon to be obsolete `polymorphic` option. 

Though there is still [other code](https://github.com/fast-jsonapi/fast_jsonapi/pull/32/files#diff-a413c6e16d70896fa04f5fa41931cadaR165-R168) that gives `polymorphic` a higher priority, so maybe it was on purpose like this (even though no other spec breaks) ... 

Maybe @christophersansone can give us some insight what should be correct or if the failing spec I provided is wrong.

Reply:
Actually [I just read](https://github.com/fast-jsonapi/fast_jsonapi/pull/32/files#r363812525) ...

> As stated earlier, if options[:serializer] is specified, then it takes precedence over any other serializer determination such as polymorphic

... which is definitely not the case here.